app/models/rbac.rb

```diff
@@ -233,38 +235,32 @@ def self.get_managed_filter_object_ids(klass, scope, filter)
     scope.find_tags_by_grouping(filter, :ns => '*', :select => minimum_columns_for(klass)).reorder(nil).collect(&:id)
   end
 
-  def self.find_targets_with_direct_rbac(klass, scope, rbac_filters, find_options = {}, user_or_group = nil)
-    u_filtered_ids = get_self_service_object_ids(user_or_group, klass)
+  def self.find_targets_with_direct_rbac(klass, scope, rbac_filters, find_options, user, miq_group)
+    u_filtered_ids = get_self_service_object_ids(user, miq_group, klass)
     b_filtered_ids = get_belongsto_filter_object_ids(klass, rbac_filters['belongsto'])
     m_filtered_ids = get_managed_filter_object_ids(klass, scope, rbac_filters['managed'])
-    d_filtered_ids = rbac_filters['ids_via_descendants']
+    d_filtered_ids = ids_via_descendants(rbac_class(klass), rbac_filters['match_via_descendants'], :user => user, :miq_group => miq_group)
 
     find_targets_filtered_by_ids(klass, scope, find_options, u_filtered_ids, b_filtered_ids, m_filtered_ids, d_filtered_ids)
   end
 
-  def self.find_targets_with_user_group_rbac(klass, scope, _rbac_filters, find_options = {}, user_or_group = nil)
-    if user_or_group && user_or_group.self_service?
-      if klass == User && user_or_group.kind_of?(User)
-        cond = ["id = ?", user_or_group.id]
-      elsif klass == MiqGroup
-        group_id = user_or_group.id if user_or_group.kind_of?(MiqGroup)
-        group_id ||= user_or_group.current_group_id if user_or_group.kind_of?(User) && user_or_group.current_group
-        cond = ["id = ?", group_id] if group_id
-      else
-        cond = nil
-      end
+  def self.find_targets_with_user_group_rbac(klass, scope, _rbac_filters, find_options, user, miq_group)
```

Comment:
yea - that has me concerned.
I am unsure why we are ignoring the scope and using klass instead